cmp/form/FormModel.ts

```diff
@@ -326,20 +332,29 @@ export class FormModel extends HoistModel {
                     // Use a regex matcher to tests for dates and format matches accurately.
                     this.setValues(
                         fieldNameMap.reduce((acc, name) => {
-                            if (
-                                isString(formValues[name]) &&
-                                formValues[name].match(/^\d{4}-\d{2}-\d{2}$/)
-                            ) {
-                                return {...acc, [name]: LocalDate.from(formValues[name])};
-                            }
-                            return {...acc, [name]: formValues[name]};
+                            return {...acc, [name]: this.processFormValue(formValues[name])};
                         }, {} as PlainObject)
                     );
                 }
             },
             owner: this
         });
     }
+
+    /**
+     * TODO
+     */
+    private processFormValue(persistedValue: PlainObject) {
+        if (isNil(persistedValue.value) || persistedValue.value === '') return null;
```

Comment:
Is a user enters a value then clears it, we dont want to populate the non-result.